### ISSUE TITLE
Support gzipped files in XMLHttpRequest file system

### ISSUE THREAD

**Issue by jvilk** [jvilk/BrowserFS#101]
Support:
* Having a gzipped `listings.json`.
* Having gzipped items in a file system. Main concern: Differentiating between compressed and uncompressed size. Browsers handle this poorly. We will see assertion failures if we are not careful. We actually will need to download the entire file before we can tell its size.
* Another issue: Unclear if fetching gzipped items will work with synchronous XHRs.

I did some tests and had some issues with the `connect` server serving up gzip files; will need to see if it is properly negotiating with the browser / figuring out the MIME type.

**Comment by jvilk** [comment]
Sidenote: This is needed for Doppio to have efficiently compressed *.jar.gz files.

**Comment by jvilk** [comment]
This does not seem possible without e.g. base64 encoding files so they are transmitted as texts. Browsers seem incapable of applying gunzip on binary data, since its usage is limited to certain MIME types associated with text.

*sigh*
